File: common/src/main/java/juuxel/adorn/platform/BlockEntityBridge.java
package juuxel.adorn.platform;

import juuxel.adorn.block.entity.BrewerBlockEntity;
import juuxel.adorn.block.entity.KitchenSinkBlockEntity;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.core.BlockPos;

public interface BlockEntityBridge {
    BrewerBlockEntity createBrewer(BlockPos pos, BlockState state);
    KitchenSinkBlockEntity createKitchenSink(BlockPos pos, BlockState state);
}

File: common/src/main/kotlin/juuxel/adorn/platform/BlockEntityBridge.kt
package juuxel.adorn.platform

import juuxel.adorn.block.entity.BrewerBlockEntity
import juuxel.adorn.block.entity.KitchenSinkBlockEntity
import net.minecraft.block.BlockState
import net.minecraft.util.math.BlockPos

interface BlockEntityBridge {
    fun createBrewer(pos: BlockPos, state: BlockState): BrewerBlockEntity
    fun createKitchenSink(pos: BlockPos, state: BlockState): KitchenSinkBlockEntity
}
